ingeniamotion/communication.py

```diff
@@ -37,6 +38,64 @@
 FILE_EXT_LFU = ".lfu"
 
 
+class RegisterUpdateCallbackModifier:
+    """Class that injects the servo alias to the register
+    update callback.
+
+     Args:
+         alias: Servo alias.
+         callback: Register update callback.
+
+    """
+
+    instances: Dict[
+        str,
+        Dict[
+            Callable[[str, Servo, Register, Union[int, float, str, bytes]], None],
+            "RegisterUpdateCallbackModifier",
+        ],
+    ] = {}
+
+    def __new__(
+        cls,
+        alias: str,
+        callback: Callable[[str, Servo, Register, Union[int, float, str, bytes]], None],
+    ) -> "RegisterUpdateCallbackModifier":
+        if alias in cls.instances:
```

Comment:
Some magic going on here. 
I would explain what is going on.

Also I think that this cache mechanism is broken if you create two motioncontrollers, since the instances list is not static instead of scoped to mc